src/main/java/org/spongepowered/api/text/Texts.java

```diff
@@ -204,17 +204,77 @@ public static Text of(Object... objects) {
                 } else {
                     // Unsupported TextAction
                 }
+
+                // Special content - does not use the specified Text formatting
+            } else if (obj instanceof TextRepresentable) {
+                builder.append(((TextRepresentable) obj).toText());
+
+            } else if (obj instanceof CatalogType) {
+                CatalogType type = (CatalogType) obj;
```

Comment:
I'm not sure about this one.
Shall this stay in `Texts.of(Object...)`
or shall `CatalogType` (or its implementations in Common)  extend `TextRepresentable`?